### DISCUSSION TITLE
How do I exclude entity A class from generating QA class?

### DISCUSSION THREAD

**Discussion by Mpp1486356697** [querydsl/querydsl#3554] [Q&A]
A uses class B(@MappedSuperclass) from the third-party jar
I have some A class, he uses the entity class (MappedSuperclass) in the third-party jar, but when generating the query class, the QB class in the third-party jar is not generated, QA has the use of QB, how should I exclude the class that uses the third-party jar or pour it into making it generate the entity class in the third-party jar.
I see APTOptions.QUERYDSL_EXCLUDED_CLASSES and QUERYDSL_INCLUDED_CLASSES how should I specify it? Or some other method? 

下面是我的pom文件，其他没有做任何的配置 springbootdataJpa+querydsl
       <dependency>
            <groupId>com.querydsl</groupId>
            <artifactId>querydsl-jpa</artifactId>
            <version>${querydsl.version}</version>
            <scope>provided</scope>
        </dependency>
        <dependency>
            <artifactId>querydsl-apt</artifactId>
            <groupId>com.querydsl</groupId>
            <version>4.3.1</version>
        </dependency>

    </dependencies>

    <build>
        <plugins>
            <plugin>
                <groupId>com.mysema.maven</groupId>
                <artifactId>apt-maven-plugin</artifactId>
                <version>1.1.3</version>
                <executions>
                    <execution>
                        <goals>
                            <goal>process</goal>
                        </goals>
                        <configuration>
                            
                            <outputDirectory>target/generated-sources/java</outputDirectory>
                            <processor>com.querydsl.apt.jpa.JPAAnnotationProcessor</processor>
                        </configuration>
                    </execution>
                </executions>
                <dependencies>
                    <dependency>
                        <groupId>com.querydsl</groupId>
                        <artifactId>querydsl-apt</artifactId>
                        <version>${querydsl.version}</version>
                    </dependency>
                </dependencies>
            </plugin>
            <plugin>
                <groupId>org.springframework.boot</groupId>
                <artifactId>spring-boot-maven-plugin</artifactId>
                <configuration>
                    <excludes>
                        <exclude>
                            <groupId>org.projectlombok</groupId>
                            <artifactId>lombok</artifactId>
                        </exclude>
                    </excludes>
                </configuration>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <configuration>
                    <source>11</source>
                    <target>11</target>
                </configuration>
            </plugin>
        </plugins>
    </build>

**Comment by Falomo** [comment]
```
 <executions>
                    <execution>
                        <phase>generate-sources</phase>
                        <goals>
                            <goal>process</goal>
                        </goals>
                        <configuration>
                            <outputDirectory>${project.build.directory}/generated-sources</outputDirectory>
                            <processor>com.querydsl.apt.jpa.JPAAnnotationProcessor</processor>
                            <options>
                                <querydsl.excludedClasses>com.example.entities.ClassName</querydsl.excludedClasses>
                            </options>
                        </configuration>
                    </execution>
                </executions>
```
